src/toast/ops/demodulation.py

```diff
@@ -0,0 +1,701 @@
+            return
+        times = demod_obs.shared[self.times]
+        for name, ivals in obs.intervals.items():
+            timespans = [[ival.start, ival.stop] for ival in ivals]
+            demod_obs.intervals[name] = IntervalList(times, timespans=timespans)
+        return
+
+    @function_timer
+    def _demodulate_flag(self, flags, wkernel, offset):
+        """Collapse flags inside the filter window and downsample"""
+        """
+        # FIXME: this is horribly inefficient but optimization may require
+        # FIXME: a compiled kernel
+        n = flags.size
+        new_flags = []
+        width = wkernel // 2 + 1
+        for i in range(0, n, self.nskip):
+            ind = slice(max(0, i - width), min(n, i + width + 1))
+            buf = flags[ind]
+            flag = buf[0]
+            for flag2 in buf[1:]:
+                flag |= flag2
+            new_flags.append(flag)
+        new_flags = np.array(new_flags)
+        """
+        # FIXME: for now, just downsample the flags.  Real data will require
+        # FIXME:    measuring the total flag within the filter window
+        flags = flags.copy()
+        # flag invalid samples in both ends
+        flags[: wkernel // 2] |= self.demod_flag_mask
+        flags[-(wkernel // 2) :] |= self.demod_flag_mask
+        new_flags = flags[offset % self.nskip :: self.nskip]
+        return new_flags
+
+    @function_timer
+    def _demodulate_common_flags(self, tod, wkernel, offset):
+        """Combine and downsample flags in the filter window"""
+        common_flags = tod.local_common_flags()
+        new_flags = self._demodulate_flag(common_flags, wkernel, offset)
+        tod.cache.put(tod.COMMON_FLAG_NAME, new_flags, replace=True)
+        return
+
+    @function_timer
+    def _demodulate_signal(self, data, obs, demod_obs, dets, lowpass):
+        """demodulate signal TOD"""
+
+        for det in dets:
+            signal = obs.detdata[self.det_data][det]
+            # Get weights
+            obs_data = data.select(obs_uid=obs.uid)
+            self.stokes_weights.apply(obs_data, dets=[det])
+            weights = obs.detdata[self.stokes_weights.weights][det]
+            # iweights = 1
+            # qweights = eta * cos(2 * psi_det + 4 * psi_hwp)
+            # uweights = eta * sin(2 * psi_det + 4 * psi_hwp)
+            iweights, qweights, uweights = weights.T
+            etainv = 1 / np.sqrt(qweights ** 2 + uweights ** 2)
+            signal_demod0 = lowpass(signal)
+            signal_demod4r = lowpass(signal * 2 * qweights * etainv)
+            signal_demod4i = lowpass(signal * 2 * uweights * etainv)
+
+            # import pdb
+            # import matplotlib.pyplot as plt
+            # plt.clf();plt.plot(lowpass(signal*2*qweights)[:100],'.');plt.plot(lowpass(signal*2*uweights)[:100],'.');plt.show()
+            # pdb.set_trace()
+
+            det_data = demod_obs.detdata[self.det_data]
+            det_data[f"demod0_{det}"] = signal_demod0
```

Comment:
Is there some reason to assign to a temporary here (and below for the 2f case), rather than just:
```
det_data[f"demod0_{det}"] = lowpass(signal)
```
?

Reply:
No reason. Removed the unnecessary lines.